File: app/src/main/java/com/mnowak/cirriculumvitae/feature/moreInfo/fragment/studiesActivities/StudiesActivitiesRecyclerViewAdapter.java
package com.mnowak.cirriculumvitae.feature.moreInfo.fragment.studiesActivities;

import androidx.databinding.DataBindingUtil;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.LinearLayout;

import com.mnowak.cirriculumvitae.R;
import com.mnowak.cirriculumvitae.databinding.ItemStudiesActivityBinding;
import com.mnowak.cirriculumvitae.databinding.ViewCheckedEventBinding;
import com.mnowak.cirriculumvitae.data.model.StudiesActivity;

import java.util.List;

public class StudiesActivitiesRecyclerViewAdapter extends RecyclerView.Adapter<StudiesActivitiesRecyclerViewAdapter.ActivityViewHolder> {

    private List<StudiesActivity> studiesActivity;
    private ItemStudiesActivityBinding itemBinding;

    StudiesActivitiesRecyclerViewAdapter(List<StudiesActivity> studiesActivity) {
        this.studiesActivity = studiesActivity;
    }

    @NonNull
    @Override
    public ActivityViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        itemBinding = ItemStudiesActivityBinding.inflate(layoutInflater, parent, false);
        return new ActivityViewHolder(itemBinding);
    }

    @Override
    public void onBindViewHolder(@NonNull ActivityViewHolder holder, int position) {
        StudiesActivity studiesActivity = this.studiesActivity.get(position);
        holder.bind(studiesActivity);
    }

    @Override
    public int getItemCount() {
        return studiesActivity.size();
    }

    @Override
    public void onViewRecycled(@NonNull ActivityViewHolder holder) {
        super.onViewRecycled(holder);
        holder.clear();
    }

    public class ActivityViewHolder extends RecyclerView.ViewHolder {
        private final ItemStudiesActivityBinding itemBinding;
        private final LayoutInflater layoutInflater;

        private LinearLayout llImportantEvents;
        private LinearLayout llRedundantEvents;

        ActivityViewHolder(ItemStudiesActivityBinding itemBinding) {
            super(itemBinding.getRoot());
            this.itemBinding = itemBinding;
            llImportantEvents = itemBinding.llImportantEvents;
            llRedundantEvents = itemBinding.llRedundantEvents;
            layoutInflater = LayoutInflater.from(itemView.getContext());
        }

        public void bind(StudiesActivity studiesActivitiesViewModel) {
            addEvents(studiesActivitiesViewModel);
            itemBinding.setViewModel(studiesActivitiesViewModel);
            itemBinding.executePendingBindings();
        }

        private void addEvents(StudiesActivity studiesActivitiesViewModel) {
            for (int i = 0; i < studiesActivitiesViewModel.getEvents().size(); i++) {
                StudiesActivity.Event event = studiesActivitiesViewModel.getEvents().get(i);
                //TODO prepare view
                LinearLayout linearLayout = chooseGroupForEvent(studiesActivitiesViewModel, i);
                bindView(studiesActivitiesViewModel, event, linearLayout);
            }
        }

        private void bindView(StudiesActivity studiesActivitiesViewModel, StudiesActivity.Event event, LinearLayout linearLayout) {
            ViewCheckedEventBinding viewCheckedEventBinding = DataBindingUtil.inflate(layoutInflater, R.layout.view_checked_event, linearLayout, true);
            viewCheckedEventBinding.setStudiesViewModel(studiesActivitiesViewModel);
            viewCheckedEventBinding.setEventViewModel(event);
            viewCheckedEventBinding.executePendingBindings();
        }

        private LinearLayout chooseGroupForEvent(StudiesActivity studiesActivitiesViewModel, int position) {
            if (studiesActivitiesViewModel.getImportantEvents() > 0 && position >= studiesActivitiesViewModel.getImportantEvents()) {
                return llRedundantEvents;
            } else {
                return llImportantEvents;
            }
        }

        public void clear() {
            llImportantEvents.removeAllViews();
            llRedundantEvents.removeAllViews();
        }
    }
}

File: app/src/main/java/com/mnowak/cirriculumvitae/feature/moreInfo/fragment/studiesActivities/StudiesActivitiesRecyclerViewAdapter.kt
package com.mnowak.cirriculumvitae.feature.moreInfo.fragment.studiesActivities

import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.LinearLayout

import com.mnowak.cirriculumvitae.R
import com.mnowak.cirriculumvitae.databinding.ItemStudiesActivityBinding
import com.mnowak.cirriculumvitae.databinding.ViewCheckedEventBinding
import com.mnowak.cirriculumvitae.data.model.StudiesActivity

class StudiesActivitiesRecyclerViewAdapter(

        private val studiesActivity: List<StudiesActivity>

) : RecyclerView.Adapter<StudiesActivitiesRecyclerViewAdapter.ActivityViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ActivityViewHolder {
        val layoutInflater = LayoutInflater.from(parent.context)
        val itemBinding = ItemStudiesActivityBinding.inflate(layoutInflater, parent, false)
        return ActivityViewHolder(itemBinding)
    }

    override fun onBindViewHolder(holder: ActivityViewHolder, position: Int) {
        val studiesActivity = this.studiesActivity[position]
        holder.bind(studiesActivity)
    }

    override fun getItemCount(): Int =
            studiesActivity.size

    override fun onViewRecycled(holder: ActivityViewHolder) {
        super.onViewRecycled(holder)
        holder.clear()
    }

    inner class ActivityViewHolder(

            private val itemBinding: ItemStudiesActivityBinding

    ) : RecyclerView.ViewHolder(itemBinding.root) {

        private val layoutInflater: LayoutInflater = LayoutInflater.from(itemView.context)

        private val llImportantEvents: LinearLayout = itemBinding.llImportantEvents
        private val llRedundantEvents: LinearLayout = itemBinding.llRedundantEvents

        fun bind(studiesActivitiesViewModel: StudiesActivity) {
            addEvents(studiesActivitiesViewModel)
            itemBinding.viewModel = studiesActivitiesViewModel
            itemBinding.executePendingBindings()
        }

        private fun addEvents(studiesActivitiesViewModel: StudiesActivity) {
            for (i in 0 until studiesActivitiesViewModel.events.size) {
                val event = studiesActivitiesViewModel.events[i]
                //TODO prepare view
                val linearLayout = chooseGroupForEvent(studiesActivitiesViewModel, i)
                bindView(studiesActivitiesViewModel, event, linearLayout)
            }
        }

        private fun bindView(studiesActivitiesViewModel: StudiesActivity, event: StudiesActivity.Event, linearLayout: LinearLayout) {
            val viewCheckedEventBinding = DataBindingUtil.inflate<ViewCheckedEventBinding>(layoutInflater, R.layout.view_checked_event, linearLayout, true)
            viewCheckedEventBinding.studiesViewModel = studiesActivitiesViewModel
            viewCheckedEventBinding.eventViewModel = event
            viewCheckedEventBinding.executePendingBindings()
        }

        private fun chooseGroupForEvent(studiesActivitiesViewModel: StudiesActivity, position: Int): LinearLayout {
            return if (studiesActivitiesViewModel.importantEvents in 1..position) {
                llRedundantEvents
            } else {
                llImportantEvents
            }
        }

        fun clear() {
            llImportantEvents.removeAllViews()
            llRedundantEvents.removeAllViews()
        }
    }
}
